modules/op-components.adoc

```diff
@@ -0,0 +1,120 @@
+// This module is included in the following assembly:
+//
+// *openshift_pipelines/understanding-openshift-pipelines.adoc
+
+:_content-type: CONCEPT
+[id="op-components_{context}"]
+= Components of {pipelines-title}
+
+{pipelines-title} is a comprehensive solution for automating continuous integration and continuous delivery (CI/CD) processes, such as building, testing, and deploying code, in a cloud-native way.
+
+With {pipelines-shortname}, you can configure the automation of any CI/CD process as a set of building blocks by using custom OpenShift Container Platform resources and creating your components that run in containers and pods.
+
+There are several ways to create your blocks and to access reusable blocks.
+
+[id="op-overview-stepstaskspipelines_{context}"]
+== Steps, tasks, and pipelines
+
+The smallest block of the Pipelines CI/CD process is a _step_. A step is a single operation, such as running one or several unit tests for a Python web application or a Java program compilation. {pipelines-title} performs each step in a separate container.
+
+A _task_, represented by a `Task` resource, is a collection of steps, which always run in a fixed sequential order. {pipelines-title} runs all steps of a task as containers in the same pod. This pod provides an environment that enables steps to access shared resources. For example, you can mount a volume in a task, and each step of the task can access this volume.
+
+A _pipeline_, represented by a `Pipeline` resource, is a collection of tasks. You can arrange tasks in a pipeline in a flexible order so some tasks can run in parallel or sequentially, depending on the pipeline definition. You can also control the execution of tasks in a pipeline in various ways, for example, specify a condition for a task to proceed, define an automatic retry if a task fails, and so on.
+
+.Example diagram of steps and tasks in a pipeline
+image::338_OpenShift_Pipeline_0523.png[A diagram representing a pipeline. Task A runs first and includes three steps. When task A completes, task B and task C both start; each of these tasks has two steps. Task D starts when both task B and task C complete; this task has one step.]
```

Comment:
This is a great explanation, but I am not sure it is fully reflected in the diagram above. Is it possible to speak with Jess Schaefer about making this a bit clearer - so that it is clear that both tasks have to complete, to start task D. Also , what happens if just on task completes but the other does not ?